pkg/flowaggregator/flowaggregator.go

```diff
@@ -482,55 +485,53 @@ func (fa *flowAggregator) fillK8sMetadata(key ipfixintermediate.FlowKey, record
 	}
 }
 
-func (fa *flowAggregator) fetchPodLabels(podAddress string) string {
-	pods, err := fa.podInformer.Informer().GetIndexer().ByIndex(podInfoIndex, podAddress)
+func (fa *flowAggregator) fetchPodLabels(podNamespace string, podName string) string {
+	pod, err := fa.podLister.Pods(podNamespace).Get(podName)
 	if err != nil {
-		klog.Warning(err)
+		klog.InfoS("Failed to get Pod", "namespace", podNamespace, "name", podName, "err", err)
 		return ""
-	} else if len(pods) == 0 {
-		klog.InfoS("No Pod objects found for Pod Address", "podAddress", podAddress)
-		return ""
-	}
-	pod, ok := pods[0].(*corev1.Pod)
-	if !ok {
-		klog.Warningf("Invalid Pod obj in cache")
 	}
 	labelsJSON, err := json.Marshal(pod.GetLabels())
 	if err != nil {
-		klog.Warningf("JSON encoding of Pod labels failed: %v", err)
+		klog.ErrorS(err, "Error when JSON encoding of Pod labels")
 		return ""
 	}
 	return string(labelsJSON)
 }
 
-func (fa *flowAggregator) fillPodLabels(key ipfixintermediate.FlowKey, record ipfixentities.Record) {
-	podLabelString := fa.fetchPodLabels(key.SourceAddress)
-	sourcePodLabelsElement, err := fa.registry.GetInfoElement("sourcePodLabels", ipfixregistry.AntreaEnterpriseID)
-	if err == nil {
-		sourcePodLabelsIE, err := ipfixentities.DecodeAndCreateInfoElementWithValue(sourcePodLabelsElement, bytes.NewBufferString(podLabelString).Bytes())
-		if err != nil {
-			klog.Warningf("Create sourcePodLabels InfoElementWithValue failed: %v", err)
-		}
-		err = record.AddInfoElement(sourcePodLabelsIE)
-		if err != nil {
-			klog.Warningf("Add sourcePodLabels InfoElementWithValue failed: %v", err)
+func (fa *flowAggregator) fillPodLabelsForSide(record ipfixentities.Record, podNamespaceIEName, podNameIEName, podLabelsIEName string) error {
+	podLabelsString := ""
+	if podName, _, exist := record.GetInfoElementWithValue(podNameIEName); exist {
+		podNameString := podName.GetStringValue()
+		if podNamespace, _, exist := record.GetInfoElementWithValue(podNamespaceIEName); exist {
+			podNamespaceString := podNamespace.GetStringValue()
+			if podNameString != "" && podNamespaceString != "" {
+				podLabelsString = fa.fetchPodLabels(podNamespaceString, podNameString)
+			}
```

Comment:
nit: the Go convention is to use `ok` (instead of `exist`) for that type of variable. The convention is followed in pkg/flowaggregator/flowrecord/record.go

Reply:
Got it, thanks.